pkg/resources/storage_integration.go

```diff
@@ -160,6 +160,10 @@ func ReadStorageIntegration(d *schema.ResourceData, meta interface{}) error {
 	// Some properties can come from the SHOW INTEGRATION call
 
 	s, err := snowflake.ScanStorageIntegration(row)
+	if err == sql.ErrNoRows {
+		// Integration not found, maybe it was already deleted and we are running a refresh
+		return err
+	}
```

Comment:
```suggestion
		if err == sql.ErrNoRows {
			// If not found, mark resource to be removed from statefile during apply or refresh
			log.Printf("[DEBUG] storage integration (%s) not found", d.Id())
			d.SetId("")
			return nil
		}
```
I think you might want something like this so we won't error out and clear the state for this resource when not found